app/src/main/java/com/bbuddies/madafaker/MadafakerFirebaseMessagingService.kt

```diff
@@ -0,0 +1,73 @@
+package com.bbuddies.madafaker
+
+import android.app.NotificationChannel
+import android.app.NotificationManager
+import android.app.PendingIntent
+import android.content.Intent
+import android.util.Log
+import androidx.core.app.NotificationCompat
+import com.bbuddies.madafaker.presentation.MainActivity
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+
+
+class MadafakerFirebaseMessagingService : FirebaseMessagingService() {
+
+    companion object {
+        private const val TAG = "MadafakerFirebaseMsgService"
+    }
+
+    override fun onMessageReceived(message: RemoteMessage) {
+        if (message.data.size > 0) {
+            Log.d(TAG, "Message Data payload: " + message.data)
+        }
+        if (message.notification != null) {
+            sendNotification(
+                message.notification!!.body, message.notification!!.title
+            )
+        }
+    }
+
+    private fun sendNotification(messageBody: String?, title: String?) {
+        val intent = Intent(this, MainActivity::class.java)
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
+        val pendingIntent = PendingIntent.getActivity(
+            this, 0 /* Request code */, intent,
+            PendingIntent.FLAG_IMMUTABLE
+        )
+
+        val channelId = getString(R.string.default_notification_channel_id)
+        val notificationBuilder: NotificationCompat.Builder =
+            NotificationCompat.Builder(this, channelId)
+                .setSmallIcon(R.drawable.ic_launcher_foreground)
+                .setContentTitle(title)
+                .setContentText(messageBody)
+                .setAutoCancel(true)
+                .setContentIntent(pendingIntent)
+        val notificationManager = getSystemService(NOTIFICATION_SERVICE) as NotificationManager
+
+        // Since android Oreo notification channel is needed.
+        val channel = NotificationChannel(
+            channelId,
+            "Channel human readable title",
+            NotificationManager.IMPORTANCE_DEFAULT
+        )
+        notificationManager.createNotificationChannel(channel)
+        notificationManager.notify(0 /* ID of notification */, notificationBuilder.build())
+    }
+
+
+    override fun onNewToken(token: String) {
+        Log.d(TAG, "Refreshed token: $token")
+
+        // If you want to send messages to this application instance or
+        // manage this apps subscriptions on the server side, send the
+        // FCM registration token to your app server.
+        sendRegistrationToServer(token)
+    }
+
+    private fun sendRegistrationToServer(token: String?) {
+        // TODO: Implement this method to send token to your app server.
```

Comment:
Create please a task on BE to implement such method, include link to Firebase doc